platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js

```diff
@@ -201,11 +201,12 @@ export class HotKeysPreferences extends Component {
               value={keys[0]}
               vali="true"
               className="form-control hotkey text-center"
+              data-cy={label}
               onKeyDown={event => this.onInputKeyDown(event, commandName)}
               onKeyUp={event => this.onChange(event, commandName)}
             />
             <span className="wrapperText" />
-            <span className="errorMessage">
+            <span className="errorMessage" data-cy={`error-${label}`}>
```

Comment:
This is tricky. I don't love `dynamic` data-cy attributes, but I'm trying to think about how else I would do this. I'll circle back after I see how you're leveraging these.

Reply:
Okay. Instead of dynamic `data-cy`, can we try...

[.closest()](https://docs.cypress.io/api/commands/closest.html#Syntax)

So, something like:

```js
// Within scopes all `.get` and `.contains` to within the matched elements
// dom instead of checking from document
cy.get('.modal-body').within(() => {

  cy.contains('Next Image Viewport') // label we're looking for
     .closest('input') // closest input to that label
     .type('c');

  cy.contains('Previous Image')
     // etc.
})
```

This should feel similar to the code you're using below to find input elements. You're still using the label, but now we're not generating dynamic cypress attributes that match text we can already seek on.

If the `data-cy` attributes made things less brittle, I would be more keen on keeping them. They're nice for "container" elements, decoupling from `class name` dependencies, etc.

`.row > .col-sm-6 > div:first-child` is much harrier than `data-cy="page-footer"`

Reply:
Thanks for this suggestion! I just had to do a small change, because `.closest()` only searches in its ancestors and the `input` is a label's sibling. Instead of that, I used `.parent().find()` and it did the trick.

```
cy.get('.modal-body').within(() => {
      cy.contains(function_label) // label we're looking for
        .parent()
        .find('input') // closest input to that label
        .type(shortcut, { force: true }); // Set new shortcut for that function
    });
```